dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/EnrollmentTimeFieldSqlRenderer.java

```diff
@@ -125,6 +126,24 @@ protected String getConditionForNonDefaultBoundaries( EventQueryParams params )
             .collect( Collectors.joining( " and " ) );
     }
 
+    private void validateProgramIndicatorBoundaries( EventQueryParams params )
+    {
+
+        boolean isValid = params.getProgramIndicator().getAnalyticsPeriodBoundaries().stream()
+            .allMatch( EnrollmentTimeFieldSqlRenderer::isValid );
+
+        if ( !isValid )
+        {
+            throw new IllegalQueryException( "Program indicator " + params.getProgramIndicator().getUid() +
```

Comment:
I believe would be better to use an ErrorCode, so clients can handle errors easier. ie.:

```
throw new IllegalQueryException( new ErrorMessage( E7136 ) );
```